proto/speechly/slu/v2beta1/batch.proto

```diff
@@ -0,0 +1,200 @@
+  // Optional.
+  Method method = 2;
+  // Possible additional headers to include in the connection.
+  // Optional.
+  repeated Header headers = 3;
+}
+
+// Describes the configuration options common for the input batch.
+message ProcessAudioBatchConfig {
+  // The processing context, Speechly Application ID.
+  // Optional. If not provided, the processing context will be determined
+  // from the login information.
+  string app_id = 1;
+  // The language(s) of the audio sent in the request as a BCP-47 language tag
+  // (e.g. "en-US"). Defaults to the target application language(s).
+  // Optional.
+  repeated string language_codes = 2;
+  // Processing configuration.
+  // Required.
+  ProcessingConfiguration processing_config = 3;
+  // Reference id for a set of related operations. For example an identifier of the source
+  // system.
+  // Optional.
+  string batch_reference = 4;
+  // Additional operation specific options.
+  // Optional.
+  repeated Option options = 5;
+}
+
+// Describes the configuration options unique to a single audio source.
+message ProcessAudioSourceRequestItem {
+  // The locator to the source audio.
+  // Required.
+  string source_url = 1;
+  // The locator to the result target. The payload will be `Operation` as JSON.
+  // Optional.
+  string destination_url = 2;
+  // HTTP endpoint to be notified on completion of the processing. The payload
+  // will be `Operation` as JSON.
+  // Optional.
+  HttpResource completion_webhook = 5;
+  // Reference id for the operation. For example an identifier of the source
+  // system.
+  // Optional.
+  string reference = 3;
+  // Additional operation specific options.
+  // Optional.
+  string device_id = 4;
+}
+
+// Describes the processing options for the audio. Note that not all options are available for
+// all languages or on all Payment Plans.
+message ProcessingConfiguration {
+  // The processing should include the token level transcription and determination of time stamps for tokens.
+  bool tokenize = 1;
+  // The processing should include translating the audio to English.
+  bool translate = 2;
+  // The processing should not include transcribing the audio to the source language. This option should be used
+  // with translate (or other similar option) to suppress the normal Speech Recognition processing.
+  bool skip_transcribe = 3;
+}
+
+// Describes a single batch operation.
+message Operation {
+  // The status of the operation.
+  enum Status {
+    // Default status is empty.
+    STATUS_INVALID = 0;
+    // The operation is queued for processing.
+    STATUS_PENDING = 1;
+    // The operation is complete and transcript is available.
+    STATUS_DONE = 2;
+    // The processing failed. Error reason is available.
+    STATUS_ERROR = 3;
+  }
+  // The id of the operation.
+  string id = 1;
+  // The reference id of the operation, if given.
+  string reference = 2;
+  // The id of the batch the operation belongs to.
+  string batch_id = 11;
```

Comment:
The gap was intentional though the continuation point was supposed to be 16 (fixed now). The smaller indexes use less space, so this will keep some in reserve for potential new "required" fields.